widgets/src/components/TaskList/TaskListContainer.js

```diff
@@ -29,36 +29,43 @@ class TaskList extends React.Component {
     columns: [],
     rows: [],
     size: 0,
-    currentPage: 0,
     connection: {},
     error: null,
     errorAlert: null,
+    lastPage: false,
   };
 
   timer = { ref: null };
 
   componentDidMount = async () => {
     const { lazyLoading, serviceUrl, pageCode, frameId } = this.props;
-    const { currentPage } = this.state;
 
-    SERVICE.URL = serviceUrl;
+    if (!LOCAL) {
+      // set the PDA domain to the URL passed via props
+      DOMAINS.PDA = serviceUrl;
+    }
 
     try {
       // config will be fetched from app-builder
-      const widgetConfigs = await getPageWidget(pageCode, frameId);
+      const widgetConfigs = await getPageWidget(pageCode, frameId, true);
       if (widgetConfigs.errors && widgetConfigs.errors.length) {
         throw widgetConfigs.errors[0];
       }
+      if (!widgetConfigs.payload) {
```

Comment:
Just a thought, this might be the case when used in app-builder - if the config is shown on drag and drop and the widget is not yet created on app-builder side. Might not be the case - don't know how app builder does that under the hood

Reply:
yeah, we can enhance this later, lez keep it for now.